src/bulls_and_cows_lib/game.cpp

```diff
@@ -24,6 +24,50 @@ namespace bulls_and_cows {
                      "    WHILE not end of game\n"
                      "    Display the board and the list of attempts so far\n"
                      "    Display a message telling if the user won or lost\n";
```

Comment:
I don't really understand what you did here.
But, instead of asking the user to give you an int, you should ask the user to give you a string.
This string is a sequence of characters. You will be able to access each of them with operator[] or iterators.

Reply:
All the following code is too complicated, don't use int for storing the code, use a std::string.
Move the code that compare the secret_code and the player's code into the body of the functions in board.cpp